core/capabilities/webapi/trigger.go

```diff
@@ -1,18 +1,267 @@
+// workflow_owners   - [OPTIONAL] list of workflow owners allowed to receive this event (affects all workflows if empty)
+// params            - key-value pairs that will be used as trigger output in the workflow Engine (translated to values.Map)
+
+type TriggerRequestPayload struct {
+	TriggerId      string     `json:"trigger_id"`
+	TriggerEventId string     `json:"trigger_event_id"`
+	Timestamp      int64      `json:"timestamp"`
+	Topics         []string   `json:"topics"`
+	Params         values.Map `json:"params"`
+}
+
+func (h *triggerConnectorHandler) HandleGatewayMessage(ctx context.Context, gatewayID string, msg *api.Message) {
+	body := &msg.Body
+	sender := ethCommon.HexToAddress(body.Sender)
+	if !h.rateLimiter.Allow(body.Sender) {
+		h.lggr.Errorw("request rate-limited")
+		return
+	}
+	if !h.allowedSendersMap[sender.String()] {
+		h.lggr.Errorw("Unauthorized Sender")
+		return
+	}
+	h.lggr.Debugw("handling gateway request", "id", gatewayID, "method", body.Method, "sender", sender)
+	var payload TriggerRequestPayload
+	err := json.Unmarshal(body.Payload, &payload)
+	if err != nil {
+		return
+	}
+
+	switch body.Method {
+	case workflow.MethodWebAPITrigger:
+		h.lggr.Debugw("added MethodWebAPITrigger message", "payload", string(body.Payload))
+		// TODO: Is the staleness check supposed to be in the gateway?
+		currentTime := time.Now()
+		// TODO: check against h.config.MaxAllowedMessageAgeSec
+		if currentTime.Unix()-3000 > payload.Timestamp {
+			// TODO: fix up with error handling update in design doc
+			response := Response{Success: false}
+			h.sendResponse(ctx, gatewayID, body, response)
+		}
+
+		// is this right?
+		// Sri did wrappedPayload, err := values.WrapMap(log.Data), does that work in this case?
+		wrappedPayload, _ := values.WrapMap(payload)
+
+		for _, trigger := range h.registeredWorkflows {
+
+			// TODO: CAPPL-24 extract the topic and then match the subscriber to this messages triggers.
+			// TODO: CAPPL-24 check the topic to see if the method is a duplicate and the trigger has been sent, ie PENDING
+			// TODO: Question asked in Web API trigger about checking for completed Triggers to return COMPLETED
+			// "TriggerEventID used internally by the Engine is a pair (sender, trigger_event_id).
+			// This is to protect against a case where two different authorized senders use the same event ID in their messages.
+
+			// TODO: how do we know PENDING state, that is node received the event but processing hasn't finished.
+			TriggerEventID := body.Sender + payload.TriggerEventId
+			tr := capabilities.TriggerResponse{
+				Event: capabilities.TriggerEvent{
+					TriggerType: triggerType,
+					ID:          TriggerEventID,
+					Outputs:     wrappedPayload, // must be *values.Map
+				},
+			}
+			trigger <- tr
+		}
+
+		// TODO: ACCEPTED, PENDING, COMPLETED
+		response := Response{Success: true, Status: "ACCEPTED"}
+		h.sendResponse(ctx, gatewayID, body, response)
```

Comment:
do we need to make this non-blocking? @bolekk had similar comment here: https://github.com/smartcontractkit/chainlink/pull/14448#discussion_r1762125129

Reply:
I don't see where your PR does the non-blocking call any more, I can't find the HTTP write.  I see a send to a channel but no more http client.  Can you link to it so I can see how you made it non blocking pls?

Reply:
I asked on your revised PR where the code is to making the http request non-blocking as I don't see it any more.